Source/player.cpp

```diff
@@ -4363,84 +4363,52 @@ void __fastcall SyncPlrAnim(int pnum)
 
 void __fastcall SyncInitPlrPos(int pnum)
 {
-	int v1; // esi
-	bool v2; // zf
-	unsigned int v3; // eax
-	int v4; // ebx
-	int v5; // edi
-	int v6; // eax
-	signed int v7; // [esp+Ch] [ebp-18h]
-	int p; // [esp+10h] [ebp-14h]
-	int v9; // [esp+14h] [ebp-10h]
-	signed int v10; // [esp+18h] [ebp-Ch]
-	signed int v11; // [esp+1Ch] [ebp-8h]
-	unsigned int i; // [esp+20h] [ebp-4h]
-	signed int v13; // [esp+20h] [ebp-4h]
+	plr[pnum]._ptargx = plr[pnum].WorldX;
+	plr[pnum]._ptargy = plr[pnum].WorldY;
 
-	p = pnum;
-	v1 = pnum;
-	v2 = gbMaxPlayers == 1;
-	plr[v1]._ptargx = plr[pnum].WorldX;
-	plr[v1]._ptargy = plr[pnum].WorldY;
-	if ( !v2 && plr[v1].plrlevel == currlevel )
-	{
-		v3 = 0;
-		for ( i = 0; ; v3 = i )
-		{
-			v4 = plr[v1].WorldX + *(int *)((char *)plrxoff2 + v3);
-			v5 = plr[v1].WorldY + *(int *)((char *)plryoff2 + v3);
-			if ( PosOkPlayer(p, v4, v5) )
-				break;
-			i += 4;
-			if ( i >= 0x20 )
-				break;
+	if ( gbMaxPlayers == 1 || plr[pnum].plrlevel != currlevel ) {
+		return;
+	}
+
+	int x;
+	int y;
+	for ( DWORD i = 0; i < 8; i++ ) {
+		x = plr[pnum].WorldX + plrxoff2[i];
+		y = plr[pnum].WorldY + plryoff2[i];
+		if ( PosOkPlayer(pnum, x, y) ) {
+			break;
 		}
-		if ( !PosOkPlayer(p, v4, v5) )
-		{
-			v11 = 0;
-			v6 = -1;
-			v13 = 1;
-			v7 = -1;
-			do
-			{
-				if ( v11 )
-					break;
-				v9 = v6;
-				while ( v6 <= v13 && !v11 )
-				{
-					v5 = v9 + plr[v1].WorldY;
-					v10 = v7;
-					do
-					{
-						if ( v11 )
-							break;
-						v4 = v10 + plr[v1].WorldX;
-						if ( PosOkPlayer(p, v10 + plr[v1].WorldX, v5) && !PosOkPortal(currlevel, v4, v5) )
-							v11 = 1;
-						++v10;
+	}
+
+	if ( !PosOkPlayer(pnum, x, y) ) {
+		BOOL posOk = FALSE;
+		int xx;
+		int yy;
+		for ( int range = 1; range < 50 && !posOk; range++ ) {
+			for ( yy = -range; yy <= range && !posOk; yy++ ) {

```

Comment:
The code here looks fine, but...
does anyone else see how unoptimized this is? Every time range grows the loop re-checks all the same positions that it did in the previous iteration, instead of the positions that only just came into range. It's not the only place in the code where this happens either, it's the same thing when trying to find a free space to drop items and such.

Reply:
I think that part of the code was not in ealier versions so probably a hasty solution